test/rest.middleware.test.js

```diff
@@ -99,6 +99,15 @@ describe('loopback.rest', function() {
       .expect(200, done);
   });
 
+  it("allows models to provide a custom HTTP path", function(done) {
+    var ds = app.dataSource('db', { connector: loopback.Memory });
+    var CustomModel = ds.createModel('CustomModel', {name: String}, {"http": { "path": "domain1/CustomModelPath" } });
+
+    app.model(CustomModel);
+    app.use(loopback.rest());
+    request(app).get(CustomModel.http.path).expect(200).end(done);
```

Comment:
Please use an explicit string instead of `CustomModel.http.path`. It is more robust, as it is testing exactly the URL that we expected. Your version is relying on LoopBack to build the path. If LoopBack ignored the http.path setting passed to `ds.createModel`, your version of this test will probably still pass.